src/Display/Slider/Slider.test.tsx

```diff
@@ -0,0 +1,311 @@
+    it(`should match snapshot when initialItem ${initialItem} is passed`, async () => {
+      const sliderRef: any = {
+        current: null,
+      };
+      const { rerender, asFragment } = render(
+        <Slider ref={sliderRef} initialItem={initialItem}>
+          <Slider.Item>a</Slider.Item>
+          <Slider.Item>b</Slider.Item>
+          <Slider.Item>c</Slider.Item>
+        </Slider>
+      );
+
+      const index = initialItem > 0 ? initialItem - 1 : 0;
+      await wait(() => {
+        expect(
+          document.querySelectorAll('li')[index].classList.contains('active')
+        ).toBeTruthy();
+      });
+
+      // mock getSliderContainerDOMNode to calculate translateX
+      sliderRef.current.getSliderContainerDOMNode = () => ({
+        getBoundingClientRect: () => ({
+          width: 500,
+        }),
+      });
+
+      rerender(
+        <Slider ref={sliderRef} initialItem={initialItem}>
+          <Slider.Item>a</Slider.Item>
+          <Slider.Item>b</Slider.Item>
+          <Slider.Item>c</Slider.Item>
+        </Slider>
+      );
+      expect(asFragment()).toMatchSnapshot();
+    });
+  };
+
+  [0, 1, 2, 3].forEach(initialItem => matchSnapshot(initialItem));
+});
+
+describe('<Slider/> prop fullContent', () => {
+  const matchSnapshot = (fullContent: boolean) => {
+    it(`should match snapshot when fullContent ${fullContent} is passed`, async () => {
+      const { asFragment } = render(
+        <SliderComponent fullContent={fullContent} />
+      );
+      expect(asFragment()).toMatchSnapshot();
+    });
+  };
+
+  [true, false].forEach(fullContent => matchSnapshot(fullContent));
+});
+
+describe('<Slider/> prop arrowWhite', () => {
+  const matchSnapshot = (arrowWhite: boolean) => {
+    it(`should match snapshot when arrowWhite ${arrowWhite} is passed`, async () => {
+      const { asFragment } = render(
+        <SliderComponent arrowWhite={arrowWhite} />
+      );
+      expect(asFragment()).toMatchSnapshot();
+    });
+  };
+
+  [true, false].forEach(arrowWhite => matchSnapshot(arrowWhite));
+});
+
+describe('<Slider/> prop removeDots', () => {
+  const matchSnapshot = (removeDots: boolean) => {
+    it(`should match snapshot when removeDots ${removeDots} is passed`, async () => {
+      const { asFragment } = render(
+        <SliderComponent removeDots={removeDots} />
+      );
+      expect(asFragment()).toMatchSnapshot();
+    });
+  };
+
+  [true, false].forEach(removeDots => matchSnapshot(removeDots));
+});
+
+describe('<Slider/> prop autoplay', () => {
+  it(`should go to next slide after 6000ms when autoplay is truthy`, () => {
+    jest.useFakeTimers();
+    const { asFragment } = render(<SliderComponent autoplay={true} />);
+
+    // validate initial active slide
+    expect(
+      document.querySelectorAll('li')[0].classList.contains('active')
+    ).toBeTruthy();
+    expect(asFragment()).toMatchSnapshot();
+
+    // validate autoplay
+    jest.advanceTimersByTime(6000);
+    expect(
```

Comment:
Should we test here that the `li` with index 0 is no longer `active`?

Reply:
ok, I will validate there is only one active li.

Reply:
oh, I validate it already using snapshot, in the snapshot only the second li has class name active.